src/main/scala/Joint.scala

```diff
@@ -47,14 +113,28 @@ object Joint {
   * @param shape A list of lines specifying the shape of the joint. Each item is a
   * 3-tuple. The first two items specify the line, while the last gives the distance
   * of the line from the joint's center in the local coordinate system.
+  * @param dipAngleParam An optional parameter that can be used to specify the dip angle for the joint.
+  *                 This avoids recalculation of a known dip angle.
+  * @param dipDirectionParam An optional parameter that can be used to specify the dip direction for
+  *                     the joint. This avoids recalculation of a known dip direction.
   */
 case class Joint(normalVec: (Double, Double, Double), localOrigin: (Double, Double, Double),
-                 center: (Double, Double, Double), dipAngle: Double, dipDirection: Double, phi: Double,
-                 cohesion: Double, shape: Seq[((Double, Double, Double),Double)]) {
+                 center: (Double, Double, Double), phi: Double, cohesion: Double,
+                 shape: Seq[((Double, Double, Double),Double)],
+                 dipAngleParam: Option[Double]=null,
```

Comment:
Did you follow the example in our current code for bounding spheres? That's a slightly different case than what we have here. For a bounding sphere, I used an `Option` because the joint could be persistent or non-persistent. If it's persistent, then the joint doesn't have a bounding sphere, so that member variable becomes `None`, otherwise the joint is non-persistent so it becomes `Some`.

In this case, any joint should have a dip angle and dip direction, so you can just use a regular `Double` instead of `Option[Double]`. Having a default value of `null` is fine, then you can just calculate the result directly like you already have below, but you don't need to wrap the result in an option.

Does this make sense?